gtsam/hybrid/GaussianMixtureFactor.cpp

```diff
@@ -28,11 +28,86 @@
 
 namespace gtsam {
 
+/**
+ * @brief Helper function to correct the [A|b] matrices in the factor components
+ * with the normalizer values.
+ * This is done by storing the normalizer value in
+ * the `b` vector as an additional row.
+ *
+ * @param factors DecisionTree of GaussianFactor shared pointers.
+ * @param varyingNormalizers Flag indicating the normalizers are different for
+ * each component.
+ * @return GaussianMixtureFactor::Factors
+ */
+GaussianMixtureFactor::Factors correct(
```

Comment:
Let’s talk about this? I had not expected this functionality here…

Reply:
This was a neat idea I had after working out the math. Now I can specify factors directly instead of calling `toFactorGraph` on a Bayes net.
Plus `toFactorGraph` is quite restrictive since it doesn't work for nonlinear functions (at least I don't see a clean way of defining nonlinear hybrid factors with how things currently are setup).

It'll come extremely handy when defining nonlinear MixtureFactors as then we can do relinearization during optimization, something that's been missing in the incumbent paradigm.

Reply:
It does not make sense to me yet. It feels it throws away some of the clarity we achieved in the document. I didn’t even understand how you get the normalization factors here. Let’s talk in person rather than here?